adding logging to capture when a trainer process is sigkilled.

aschhabra opened the pull request:
Summary:
It will help us determine when a trainer process is not terminated gracefully due to SIGKILL by torch elastic.
In case of process is terminated with SIGKILL, there is no way to collect logs before termination which causes confusion due to missing logs. Logging when trainer was SIGKILLED when help understand failure for the trainer process.

Test Plan:
unit tests

buck tests test/distributed/elastic/multiprocessing/errors/subprocess_handler_test.py
 https://www.internalfb.com/intern/testinfra/testrun/12103424072581559


**ran an e2e job to confirm that training jobs using torch elastic will continue to run gracefully.**
https://www.internalfb.com/mlhub/pipelines/runs/mast/f710719113-TrainingApplication_OYJWR?job_attempt=0&version=0&tab=execution_details&env=PRODUCTION

Differential Revision: D71296393




cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

wconstab commented:
this looks reasonable to me.  I'm not very familiar with the elastic codebase, @d4l3k could you take a look?

aschhabra committed:
adding logging to capture when a trainer process is sigkilled. (#149739)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/149739

It will help us determine when a trainer process is not terminated gracefully due to SIGKILL by torch elastic.
In case of process is terminated with SIGKILL, there is no way to collect logs before termination which causes confusion due to missing logs. Logging when trainer was SIGKILLED when help understand failure for the trainer process.

Test Plan:
unit tests

buck tests test/distributed/elastic/multiprocessing/errors/subprocess_handler_test.py
 https://www.internalfb.com/intern/testinfra/testrun/12103424072581559

**ran an e2e job to confirm that training jobs using torch elastic will continue to run gracefully.**
https://www.internalfb.com/mlhub/pipelines/runs/mast/f710719113-TrainingApplication_OYJWR?job_attempt=0&version=0&tab=execution_details&env=PRODUCTION

Differential Revision: D71296393
